go-controller/pkg/ovn/egressip.go

```diff
@@ -2099,11 +2101,39 @@ func (oc *Controller) checkEgressNodesReachability() {
 }
 
 func (oc *Controller) isReachable(node *egressNode) bool {
```

Comment:
This entire function is very confusing at this point. I guess it's fine that `EgressIPReachabiltyTimeout  = 0` means "disable it", but we should really be a) using the timeout per `dial` and b) define a constant amount of times we retry, instead of this variable computed retry count based on what the timeout is. I think this is rather confusing to read and will be confusing for users to understand (most likely leading to reported bugs with egress IP, which might just be due to a misunderstanding of what this timeout really does).

Reply:
how about use the config knob just to enable/disable the node reachability check and just do osdn alike retry with timeout @trozet @alexanderConstantinescu ? so we won't get into any of this unnecessary complication.